### ISSUE TITLE
terminate-and-replace: replace operation took more than 8 hours 

### ISSUE THREAD

**Issue by yarongilor** [scylladb/scylla-cluster-tests#5688]
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

Adding a new node during terminate-and-replace.
Adding the new node exceeded the test timeout of 8 hours.
The scenario is:
1. Terminate node-5. (< t:2023-01-14 13:03:31,872 f:nemesis.py      l:1007 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.SisyphusMonkey: Set current_disruption -> TerminateAndReplaceNode Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-5 [16.170.155.37 | 10.0.2.72] (seed: True)
)
2. Add a new node node-6: longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6
3.
``` 
< t:2023-01-14 13:15:35,733 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2023-01-14T13:15:35+00:00 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6     !INFO | scylla[7733]:  [shard  0] storage_service - Starting to bootstrap...
```
5.  Init is not finished in 8 hours so test aborted this operation: 
```
< t:2023-01-14 21:15:41,433 f:nemesis.py      l:1113 c:sdcm.nemesis         p:WARNING > sdcm.nemesis.SisyphusMonkey: TestConfig of the 'Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 [13.49.224.84 | 10.0.3.2] (seed: False)' failed, removing it from list of nodes
< t:2023-01-14 21:15:41,433 f:nemesis.py      l:1115 c:sdcm.nemesis         p:WARNING > sdcm.nemesis.SisyphusMonkey: Node will not be terminated. Please terminate manually!!!
< t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG >     wait.wait_for(func=self.db_up, step=60, text=text, timeout=timeout, throw_exc=True)
< t:2023-01-14 21:16:41,490 f:nemesis.py      l:1102 c:sdcm.nemesis         p:INFO  > sdcm.nemesis.SisyphusMonkey: Adding new node to cluster...

```
6. Init finished only 15 minutes later:
```
< t:2023-01-14 21:30:12,953 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2023-01-14T21:30:12+00:00 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6     !INFO | scylla[7733]:  [shard  0] storage_service - Bootstrap completed! 
```
logs:
```
&lt; t:2023-01-14 21:15:41,433 f:nemesis.py      l:1113 c:sdcm.nemesis         p:WARNING &gt; sdcm.nemesis.SisyphusMonkey: TestConfig of the 'Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 [13.49.224.84 | 10.0.3.2] (seed: False)' failed, removing it from list of nodes
&lt; t:2023-01-14 21:15:41,433 f:nemesis.py      l:1115 c:sdcm.nemesis         p:WARNING &gt; sdcm.nemesis.SisyphusMonkey: Node will not be terminated. Please terminate manually!!!
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; '_add_and_init_new_cluster_node': failed with '[Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 [13.49.224.84 | 10.0.3.2] (seed: False)] NodeSetupFailed: RetryError[Wait for: db_up: timeout - 28800 seconds - expired]
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; Traceback (most recent call last):
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 64, in wait_for
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     res = retry(func, **kwargs)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 404, in __call__
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     do = self.iter(retry_state=retry_state)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 360, in iter
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise retry_exc.reraise()
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 194, in reraise
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise self
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; tenacity.RetryError: RetryError[&lt;Future at 0x7fd9e5dbfd00 state=finished returned bool&gt;]
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; 
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; The above exception was the direct cause of the following exception:
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; 
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; Traceback (most recent call last):
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3506, in node_setup
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     cl_inst.node_setup(_node, **setup_kwargs)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 4235, in node_setup
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     node.wait_db_up(verbose=verbose, timeout=timeout)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 1297, in wait_db_up
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     wait.wait_for(func=self.db_up, step=60, text=text, timeout=timeout, throw_exc=True)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 75, in wait_for
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise tenacity.RetryError(err) from ex
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; tenacity.RetryError: RetryError[Wait for: db_up: timeout - 28800 seconds - expired]
```

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Kernel Version: 5.15.0-1028-aws
Scylla version (or git commit hash): `2022.2.0-20230112.4f0f82ff2e1d` with build-id `a1d0fdd6fed1f802854040dbefcf5c45bbc4e820`

Cluster size: 4 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-9 (13.49.74.30 | 10.0.3.207) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-8 (13.51.69.138 | 10.0.1.55) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 (13.48.178.220 | 10.0.3.40) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 (13.49.224.84 | 10.0.3.2) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-5 (16.170.155.37 | 10.0.2.72) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-4 (13.49.73.236 | 10.0.0.61) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-3 (16.16.28.144 | 10.0.1.215) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-2 (13.49.223.113 | 10.0.2.243) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-11 (13.49.57.210 | 10.0.3.116) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-10 (13.49.73.2 | 10.0.2.154) (shards: 14)
- longevity-tls-1tb-7d-2022-2-db-node-97b904bb-1 (13.51.69.33 | 10.0.3.112) (shards: 14)

OS / Image: `ami-098f27180ca4e2195` (aws: eu-north-1)

Test: `ics-longevity-1tb-7days-test`
Test id: `97b904bb-2a97-4646-ab99-cc22b545cb44`
Test name: `enterprise-2022.2/SCT_Enterprise_Features/ICS/ics-longevity-1tb-7days-test`
Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/1fede40e24b751c5ce1e84883885fe7be765afa5/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 97b904bb-2a97-4646-ab99-cc22b545cb44`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=97b904bb-2a97-4646-ab99-cc22b545cb44)
- Show all stored logs command: `$ hydra investigate show-logs 97b904bb-2a97-4646-ab99-cc22b545cb44`

## Logs:
*No logs captured during this run.*

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.2/job/SCT_Enterprise_Features/job/ICS/job/ics-longevity-1tb-7days-test/9/)
</details>

**Comment by fruch** [comment]
@yarongilor why aren't you stopping this run ? it was doing one nemesis for 16h ?

**Comment by yarongilor** [comment]
The issue is not reproduced with a reproducer. The nemesis passed ok. Test-id: 195cb503-3a85-47d4-8618-6f5188956094

**Comment by yarongilor** [comment]
> @yarongilor why aren't you stopping this run ? it was doing one nemesis for 16h ?

@fruch , from some reason the grafana failed to open so i didn't stop it. Anyway the test failed in a short while since didn't have enough nodes.
Another problem is that the pipeline steps of:
`Collect log data | Clean resources | Send email with result`
all failed as well.

**Comment by yarongilor** [comment]
Log were manually collected.
Logs from node-7 had some errors like:
```
Jan 17 08:04:54 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71155]: Invalid user sFTPUser from 83.100.164.237 port 37509
Jan 17 08:04:55 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71155]: Connection reset by invalid user sFTPUser 83.100.164.237 port 37509 [preauth]
Jan 17 08:05:19 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71158]: Invalid user admin from 211.75.215.145 port 54713
Jan 17 08:05:25 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71158]: error: maximum authentication attempts exceeded for invalid user admin from 211.75.215.145 port 54713 ssh2 [preauth]
Jan 17 08:05:25 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71158]: Disconnecting invalid user admin 211.75.215.145 port 54713: Too many authentication failures [preauth]
Jan 17 08:05:29 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71161]: Invalid user admin from 211.75.215.145 port 54796
Jan 17 08:05:30 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 sshd[71161]: Connection reset by invalid user admin 211.75.215.145 port 54796 [preauth]
```
scylla server did start ok when started it manually:
```
scyllaadm@longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7:~$ sudo systemctl start scylla-server
scyllaadm@longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7:~$ systemctl status scylla-server
● scylla-server.service - Scylla Server
     Loaded: loaded (/lib/systemd/system/scylla-server.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/scylla-server.service.d
             └─capabilities.conf, dependencies.conf, mounts.conf, override.conf, sysconfdir.conf
     Active: active (running) since Tue 2023-01-17 08:14:06 UTC; 7s ago
    Process: 71399 ExecStartPre=/opt/scylladb/scripts/scylla_prepare (code=exited, status=0/SUCCESS)
   Main PID: 71508 (scylla)
     Status: "starting sstables loader"
      Tasks: 2 (limit: 147276)
     Memory: 1.6G
     CGroup: /scylla.slice/scylla-server.slice/scylla-server.service
             └─71508 [scylla]

Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down API server was successful
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down tracing instance
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down tracing instance was successful
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down migration manager notifier
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down migration manager notifier was successful
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down prometheus API server
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down prometheus API server was successful
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down sighup
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Shutting down sighup was successful
Jan 17 08:14:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-7 scylla[71508]:  [shard  0] init - Startup failed: std::runtime_error (Cannot replace_address 10.0.2.72 because it doesn't exist in gossip)
```

Full logs:
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                            Log links for testrun with test id 97b904bb-2a97-4646-ab99-cc22b545cb44                                                                                                            |
+-----------------+-------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                                                          |
+-----------------+-------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20230114_130840 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230114_130840/grafana-screenshot-ics-longevity-1tb-7days-test-scylla-per-server-metrics-nemesis-20230114_130952-longevity-tls-1tb-7d-2022-2-monitor-node-97b904bb-1.png |
| 20230114_130840 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230114_130840/grafana-screenshot-overview-20230114_130840-longevity-tls-1tb-7d-2022-2-monitor-node-97b904bb-1.png                                                       |
| 20230115_083219 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230115_083219/grafana-screenshot-ics-longevity-1tb-7days-test-scylla-per-server-metrics-nemesis-20230115_083255-longevity-tls-1tb-7d-2022-2-monitor-node-97b904bb-1.png |
| 20230115_083219 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230115_083219/grafana-screenshot-overview-20230115_083219-longevity-tls-1tb-7d-2022-2-monitor-node-97b904bb-1.png                                                       |
| 20230116_153745 | critical    | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/critical-97b904bb.log.tar.gz                                                                                                                              |
| 20230116_153745 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/db-cluster-97b904bb.tar.gz                                                                                                                                |
| 20230116_153745 | debug       | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/debug-97b904bb.log.tar.gz                                                                                                                                 |
| 20230116_153745 | error       | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/error-97b904bb.log.tar.gz                                                                                                                                 |
| 20230116_153745 | event       | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/events-97b904bb.log.tar.gz                                                                                                                                |
| 20230116_153745 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/loader-set-97b904bb.tar.gz                                                                                                                                |
| 20230116_153745 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/monitor-set-97b904bb.tar.gz                                                                                                                               |
| 20230116_153745 | normal      | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/normal-97b904bb.log.tar.gz                                                                                                                                |
| 20230116_153745 | output      | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/output-97b904bb.log.tar.gz                                                                                                                                |
| 20230116_153745 | event       | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/raw_events-97b904bb.log.tar.gz                                                                                                                            |
| 20230116_153745 | sct         | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/sct-97b904bb.log.tar.gz                                                                                                                                   |
| 20230116_153745 | summary     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/summary-97b904bb.log.tar.gz                                                                                                                               |
| 20230116_153745 | warning     | https://cloudius-jenkins-test.s3.amazonaws.com/97b904bb-2a97-4646-ab99-cc22b545cb44/20230116_153745/warning-97b904bb.log.tar.gz                                                                                                                               |
+-----------------+-------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

**Comment by fruch** [comment]
@yarongilor why are you using SCT issue as a notepad, i.e. there no one single information here that point to any issue with SCT.

and you say scylla started o.k. manually while the logs there say:
`Startup failed: std::runtime_error (Cannot replace_address 10.0.2.72 because it doesn't exist in gossip)`

**Comment by fruch** [comment]
I think your change in https://github.com/scylladb/scylla-cluster-tests/pull/5426 wasn't enough

you have here a replace that took 8h:15m (while you extended the duration to 8h total, which includes adding the new node)

```
Jan 14 13:15:35 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 scylla[7733]:  [shard  0] storage_service - replace[edd046e0-d256-439d-aa2f-c7168a91ab5d]: Started replace operation, replace_nodes={{ 10.0.2.72 -> 10.0.3.2}}, sync_nodes=[10.0.3.2, 10.0.0.61, 10.0.3.112, 10.0.1.215], ignore_nodes=[]
...
Jan 14 21:30:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 scylla[7733]:  [shard  0] storage_service - replace[edd046e0-d256-439d-aa2f-c7168a91ab5d]: Allow any nodes to mark replacing node=10.0.3.2 as alive
```

I think `replace_node` should have it's own duration (not use the default)
```
    @latency_calculator_decorator(legend="Replace a node in cluster with new one")
    def replace_node(self, old_node_ip, rack=0):
        return self._add_and_init_new_cluster_node(old_node_ip, rack=rack)
```

and also should consider using SoftTimeout for this operation, like 9h soft, 12 hour hard.

https://github.com/fruch/scylla-cluster-tests/commit/2a9b998e7b77e0d1ee188053a2abef02d1e1f86b

Anyhow we shouldn't ever try to retry replace operation, since it there's ongoing replacement, any retry would fail.

**Comment by yarongilor** [comment]
> I think your change in #5426 wasn't enough
> 
> you have here a replace that took 8h:15m (while you extended the duration to 8h total, which includes adding the new node)
> 
> ```
> Jan 14 13:15:35 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 scylla[7733]:  [shard  0] storage_service - replace[edd046e0-d256-439d-aa2f-c7168a91ab5d]: Started replace operation, replace_nodes={{ 10.0.2.72 -> 10.0.3.2}}, sync_nodes=[10.0.3.2, 10.0.0.61, 10.0.3.112, 10.0.1.215], ignore_nodes=[]
> ...
> Jan 14 21:30:11 longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 scylla[7733]:  [shard  0] storage_service - replace[edd046e0-d256-439d-aa2f-c7168a91ab5d]: Allow any nodes to mark replacing node=10.0.3.2 as alive
> ```
> 
> I think `replace_node` should have it's own duration (not use the default)
> 
> ```
>     @latency_calculator_decorator(legend="Replace a node in cluster with new one")
>     def replace_node(self, old_node_ip, rack=0):
>         return self._add_and_init_new_cluster_node(old_node_ip, rack=rack)
> ```
> 
> and also should consider using SoftTimeout for this operation, like 9h soft, 12 hour hard.
> 
> [fruch@2a9b998](https://github.com/fruch/scylla-cluster-tests/commit/2a9b998e7b77e0d1ee188053a2abef02d1e1f86b)
> 
> Anyhow we shouldn't ever try to retry replace operation, since it there's ongoing replacement, any retry would fail.

right, i just updated it in the Argus job and not here:
The following error happened previously in this nemesis:
```
&lt; t:2023-01-14 21:15:41,433 f:nemesis.py      l:1113 c:sdcm.nemesis         p:WARNING &gt; sdcm.nemesis.SisyphusMonkey: TestConfig of the 'Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 [13.49.224.84 | 10.0.3.2] (seed: False)' failed, removing it from list of nodes
&lt; t:2023-01-14 21:15:41,433 f:nemesis.py      l:1115 c:sdcm.nemesis         p:WARNING &gt; sdcm.nemesis.SisyphusMonkey: Node will not be terminated. Please terminate manually!!!
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; '_add_and_init_new_cluster_node': failed with '[Node longevity-tls-1tb-7d-2022-2-db-node-97b904bb-6 [13.49.224.84 | 10.0.3.2] (seed: False)] NodeSetupFailed: RetryError[Wait for: db_up: timeout - 28800 seconds - expired]
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; Traceback (most recent call last):
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 64, in wait_for
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     res = retry(func, **kwargs)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 404, in __call__
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     do = self.iter(retry_state=retry_state)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 360, in iter
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise retry_exc.reraise()
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/tenacity/__init__.py", line 194, in reraise
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise self
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; tenacity.RetryError: RetryError[&lt;Future at 0x7fd9e5dbfd00 state=finished returned bool&gt;]
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; 
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; The above exception was the direct cause of the following exception:
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; 
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; Traceback (most recent call last):
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3506, in node_setup
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     cl_inst.node_setup(_node, **setup_kwargs)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 4235, in node_setup
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     node.wait_db_up(verbose=verbose, timeout=timeout)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 1297, in wait_db_up
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     wait.wait_for(func=self.db_up, step=60, text=text, timeout=timeout, throw_exc=True)
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;   File "/home/ubuntu/scylla-cluster-tests/sdcm/wait.py", line 75, in wait_for
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt;     raise tenacity.RetryError(err) from ex
&lt; t:2023-01-14 21:15:41,434 f:decorators.py   l:69   c:sdcm.utils.decorators p:DEBUG &gt; tenacity.RetryError: RetryError[Wait for: db_up: timeout - 28800 seconds - expired]
```

**Comment by fgelcer** [comment]
@asias , does that make sense that replace node operation is taking more than 8 hours (this last failure, it took 8 hours and 15 minutes)?

https://github.com/scylladb/scylla-cluster-tests/issues/5688#issuecomment-1385170386

**Comment by fgelcer** [comment]
@yarongilor , please send a PR to increase the replace operation timeout

**Comment by asias** [comment]
It really depends on the workload and data volume. One thing to note is
that we now do off strategy compaction as soon as a table is finished
during replace, the compaction would slow down the streaming. The total
time increases but we finish the compaction sooner.

**Comment by yarongilor** [comment]
Addressed in #2600
